libcaf_core/caf/node_id.cpp

```diff
@@ -95,29 +95,18 @@ bool hashed_node_id::can_parse(std::string_view str) noexcept {
 
 node_id hashed_node_id::local(const actor_system_config&) {
   CAF_LOG_TRACE("");
-  auto ifs = detail::get_mac_addresses();
-  std::vector<std::string> macs;
-  macs.reserve(ifs.size());
-  for (auto& i : ifs)
-    macs.emplace_back(std::move(i.second));
-  auto seeded_hd_serial_and_mac_addr = join(macs, "") + detail::get_root_uuid();
-  // By adding 8 random ASCII characters, we make sure to assign a new (random)
-  // ID to a node every time we start it. Otherwise, we might run into issues
-  // where a restarted node produces identical actor IDs than the node it
-  // replaces. Especially when running nodes in a container, because then the
-  // process ID is most likely the same.
+  // We add an global incrementing counter to make sure two actor systems in the
+  // same process won't have the same node ID - even if the user manipulates the
+  // system to always produce the same seed for its randomness.
+  auto sys_seed = static_cast<uint8_t>(system_id.fetch_add(1));
```

Comment:
What is bogging me is that every time we call the `local` function we get a new value. So is this part still needed?

Reply:
Hm, good question. The chance that `random_device` returns the same value is probably very small.

Reply:
There's the possibility from the comment -  the user manipulating the system to always produce the same seed. Then again, if I'm the user that manipulates the system to produce the same seed, I probably want the same random numbers generated.